### PR TITLE
Added docs about sidecar installation

### PR DESCRIPTION
**Pull request by fawadkhaliq**
**Issue** #350 

**Description of changes**
Added documentation on sidecar/envoy installation for virtual nodes and virtual gateways

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: docs/reference/injector.md [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+
+App Mesh uses namespace and/or pod annotations to determine if pods in a namespace will be marked for sidecar injection. There are two ways to achieve this:
+
+`appmesh.k8s.aws/sidecarInjectorWebhook: enabled`: The sidecar injector will inject the sidecar into pods by default. Add the `appmesh.k8s.aws/sidecarInjectorWebhook` annotation with value `disabled` to the pod template spec to override the default and disable injection. For example:
+
+```
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: default-disabled
```
Comment (achevuru) [inline docs/reference/injector.md:19]: Should this be 'default-enabled'? I see the below deployment is tied to 'default-enabled'.

File: docs/reference/injector.md [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+  labels:
+    mesh: default-disabled
+    appmesh.k8s.aws/sidecarInjectorWebhook: disabled
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: default-behavior // sidecar will be not injected as namespace has sidecar injection disabled
+  namespace: default-enabled
```
Comment (achevuru) [inline docs/reference/injector.md:67]: Same comment as above, w.r.t namespace name...

File: docs/reference/injector.md [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+      - name: override-default-disabled
+        image: tutum/curl
+```
+
+
+## Envoy injection for virtual gateways
+
+AWS App Mesh supports virtual gateway resource to allow resources that are outside of your mesh to communicate to resources that are inside of your mesh. The virtual gateway represents an Envoy proxy running in the Kubernetes cluster. Unlike a virtual node, which represents Envoy running with an application, a virtual gateway represents Envoy deployed by itself. App Mesh Kubernetes controller supports injecting Envoy and virtual gateway configuration.
+
+App Mesh Kubernetes controller uses podSelector to designate Virtual Gateway membership. If you create an pod with labels matching the pod selector labels in a virtual gateway spec, the controller will inject the Envoy configuration to the pod/envoy container and override the default container image by default.
```
Comment (achevuru) [inline docs/reference/injector.md:96]: nit: 'create a pod'

File: docs/reference/injector.md [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+```
+apiVersion: appmesh.k8s.aws/v1beta2
+kind: VirtualGateway
+metadata:
+  name: ingress-gw
+  namespace: ns
+spec:
+  namespaceSelector:
+    matchLabels:
+      gateway: ingress-gw
```
Comment (achevuru) [inline docs/reference/injector.md:111]: We're missing this in above VirtualGateway description (i.e.,) the need to have a namespace label for VG.

File: docs/reference/injector.md (hunk 1 above)
Comment (fawadkhaliq) [reply docs/reference/injector.md:19]: Yess!! let me fix this

File: docs/reference/injector.md (hunk 4 above)
Comment (fawadkhaliq) [reply docs/reference/injector.md:111]: Done, added to the description

**Review by achevuru — APPROVED** [review]

**Pull request merged**